### PR TITLE
Feat/task8

### PR DESCRIPTION
**Pull request by dartwlad**
EB: http://dartwlad-cart-api.us-east-1.elasticbeanstalk.com/api/profile/cart
FE APP: https://d2igrds53f0okp.cloudfront.net/  (PROBLEMS WITH HTTPS, WORKS ON LOCALHOST AS EXPECTED)
FE PR: https://github.com/dartwlad/nodejs-aws-fe/pull/8
image

Run docker lint:
npm run lint:docker

 1 - Dockerfile is prepared, image is building. Image size is minimised to be less than 500 MB.
 2 - Dockerfile is optimized. Files that change more often and commands that depend on them should be included later, files and commands that change less should be at the top.
 3 - Folders are added to .dockerignore, with explanations. At least 2 big directories should be excluded from build context. Elastic Beanstalk application is initialized.
 4 - Environment is created and the app is deployed to the AWS cloud. You must provide a link to your GitHub repo with Cart API service or PR with created Dockerfile and related configurations.
 5 - FE application is updated with Cart API endpoint. You must provide a PR with updates in your FE repository and OPTIONALLY link to deployed front-end app which makes proper API calls to your Cart service.

### CODE REVIEW

File: Dockerfiles/Dockerfile [hunk 1]
```diff
@@ -0,0 +1,28 @@
...
+LABEL Name="dartwlad-cart-api"
+LABEL Version="0.0.1"
+
+# Dependencies
+WORKDIR /app
+COPY package*.json ./
+RUN npm i
+
+# Build
+WORKDIR /app
```
Comment (aleksei-bulgak-study) [inline Dockerfiles/Dockerfile:12]: I suppose you do not need second `WORKDIR /app` as you specified it at line 7

File: Dockerfiles/Dockerfile [hunk 2]
```diff
@@ -0,0 +1,28 @@
...
+COPY . .
+RUN npm run build
+
+# Application
+FROM node:12-alpine as application
+
+COPY --from=base /app/package*.json ./
+RUN npm i --only=production
+RUN npm i -g pm2
```
Comment (aleksei-bulgak-study) [inline Dockerfiles/Dockerfile:21]: cool idea to use `pm2`. I have never use it but I heard it is really cool tool

File: scripts/docker/build-and-publish.sh [hunk 3]
```diff
@@ -0,0 +1,28 @@
+#!/bin/bash
+
+REGION=us-east-1
+AWS_ACCOUNT_ID=443280315061
```
Comment (aleksei-bulgak-study) [inline scripts/docker/build-and-publish.sh:4]: Accaunt id is secure information and can be used to hack your account. Please try to use env variables and not to commit it in VCS

**Comment by andech-epam** [comment]
It's not necessary, but there is an easy way to add https support to your beanstalk app https://nickvernij.nl/blog/setup-https-docker-elastic-beanstalk.html, please, check.

**Comment by dartwlad** [comment]
> It's not necessary, but there is an easy way to add https support to your beanstalk app https://nickvernij.nl/blog/setup-https-docker-elastic-beanstalk.html, please, check.

Thanks!
Will try.

**Pull request merged**